### ISSUE TITLE
Is mask decoder trained in the second stage?

### ISSUE THREAD

**Issue by lx709** [zamling/PSALM#15]
Hello @zamling, it's mentioned that the mask decoder is trained in the paper. However, I noticed that in the code is_train_mask_decode is set to False by default, this means the mask decoder is not trained. I'm wondering which one should I use for my own dataset. Thanks for your attention.

**Comment by zamling** [comment]
Hi @lx709 
The name, 'is_train_mask_decode', may be ambiguous. It means whether the mask decoder have been trained.
As the model initialized by LLM ckpt, mask decoder is not initailized by `from_pretrained`. The mask decoder will be initialized here
https://github.com/zamling/PSALM/blob/025fa75403fba97a50a857170396cbd0aec077e7/psalm/train/train.py#L378-L380

And 'is_train_mask_decode' will be set True after initializing, so that when inference or finetune based on another ft ckpt, the mask decoder can be loaded by from_pretrained. As shown below: 
https://github.com/zamling/PSALM/blob/025fa75403fba97a50a857170396cbd0aec077e7/psalm/model/language_model/llava_phi.py#L170-L173

https://github.com/zamling/PSALM/blob/025fa75403fba97a50a857170396cbd0aec077e7/psalm/model/language_model/llava_phi.py#L165-L167

**Comment by lx709** [comment]
@zamling , thanks for your clarification.
